### PR TITLE
Issue 22533 relaystate

### LINKED ISSUE #22533: SAML Improvements
This is a list to capture a number of improvements that can be made to our dotSAML implementation

- #22156 redirect is not working for front end login.  Maybe we need to add a redirect after login field to our saml secrets and use that if a redirect is not provided?

> We figured this out - the issue was that we needed to hit the SAMLWebInterceptor in order to login.  What needs to happen is to add `include.path.values=*`  where * is a regex for the path where you want to force/redirect your saml logins and you add  that `access.filter.values=/api` to exclude /api calls from redirecting to SAML.  You can see a working example on our intranet 

- #21430 roles causing an NPE
- Make sure that the "Service Provider Endpoint Hostname/Port" field is normalized - this can be done under the covers 
   - The field should only expect `hostname:port` here, where port is optional.
   - The value should not have `http://` or `https://` in it as we will automatically prepend `https`
   - we should automatically add the whole dotSAML path to whatever is provided here `/api/v1/dotsaml/login`
- Fix the `MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=application/text` errors that eat/obfuscate any original SAML error.   My guess is we can add the `jersey-media-json-jackson` to the dotSAML plugin or make sure it is exported?
- Try to map common fields without configuration.  We should have a list of aliases for each field that we expect back from the idp and try to use those before giving up
- #22124 this does not work if the user has logged in using SAML
- We need to try to log the user in on the /api/v1/dotsaml/login/* endpoint.  That way, when they redirect, they will already be logged in.

### PR DESCRIPTION
**Pull request by jdotcms**
Adding the relay state (core part) on SAML

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebUtils.java [hunk 1]
```diff
@@ -185,4 +214,78 @@ protected HttpSession renewSession(final HttpServletRequest request, final HttpS
...
+     * @param identityProviderConfiguration
+     * @param siteIdentifier
+     * @return
+     */
+    public String getRelayState(final HttpServletRequest request,
+                                final HttpServletResponse response,
+                                final IdentityProviderConfiguration identityProviderConfiguration,
+                                final String siteIdentifier) {
+
+        final String relayState = identityProviderConfiguration.containsOptionalProperty(AUTH_RELAYSTATE_KEY)?
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebUtils.java:234]: null check on identityProviderConfiguration

File: dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebUtils.java [hunk 2]
```diff
@@ -185,4 +214,78 @@ protected HttpSession renewSession(final HttpServletRequest request, final HttpS
...
+                                final String relayStateTemplate,
+                                final String siteIdentifier) {
+
+        request.setAttribute(AUTH_RELAYSTATE_KEY, relayStateTemplate);
+
+        final RelayStateStrategy relayStateStrategy = RELAY_STATE_STRATEGY_MAP.getOrDefault(siteIdentifier, DEFAULT_VELOCITY_RELAY_STATE_STRATEGY);
+
+        final String relayStateValue =  relayStateStrategy.apply(request, response, identityProviderConfiguration);
+
+        Optional.ofNullable(request.getSession(false)).ifPresent(session -> session.setAttribute(AUTH_RELAYSTATE_KEY, relayStateValue));
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebUtils.java:263]: Shouldn't we always set it here?  meaning, create the session if it does not exist at this point?  the user is in the middle of a login

**Review by wezell — COMMENTED** [review]
Looks good - see comments.

**Review by wezell — APPROVED** [review]
Looks good.

**Review by jcastro-dotcms — APPROVED** [review]

**Pull request merged**